thread_timeout/__init__.py

```diff
@@ -101,84 +103,145 @@ class NotKillExecTimeoutException(ExecTimeoutException):
+    def func(delay):
+        print("  .. sleeing for %s" % delay)
         time.sleep(delay)
-        print ("  .. done sleep for %s" % delay)
+        print("  .. done sleep for %s" % delay)
     try:
-        res = test1(1)
-        print ("Test1 OK")
+        res = func(1)
+        print("Test1 OK")
     except ExecTimeoutException:
-        print ("Test1 failed, timeout too soon")
+        print("Test1 failed, timeout too soon")
+
 
+def test2():
+    ''' timeout is stopping long function
+    '''
     @thread_timeout(1)
-    def test2(delay):
-        print ("  .. sleeing for %s" % delay)
+    def func(delay):
+        print("  .. sleeing for %s" % delay)
         time.sleep(delay)
         print("  .. done sleep for %s" % delay)
 
     try:
-        test2(3)
+        func(3)
         raise Exception("Test2 failed: timeout does not work")
     except ExecTimeoutException as e:
-        print ("  .. got excepted execption %s" % repr(e))
+        print("  .. got excepted execption %s" % repr(e))
         print("Test2 OK")
 
+
+def test3():
+    ''' function returns result
+    '''
     @thread_timeout(1)
-    def test3(x):
+    def func(x):
         return x
 
-    if test3('OK') != 'OK':
-        print ("Test3 failed, %s", repr(test3('OK')))
-    else:
-        print ("Test3 OK")
+    assert func('OK') == 'OK'
 
+
+def test4():
+    ''' FailedKillExecTimeoutException
+    FIXME! This is a wired test.  thread_timeout should actually stop this function 
+    '''
     @thread_timeout(1)
-    def test4(x):
+    def looong(x):
         for a in range(0, x):
             time.sleep(2)
     try:
-        test4(2)
+        looong(20)
+        raise Exception('FailedKillExecTimeoutException was expected')
     except FailedKillExecTimeoutException as e:
-        print ("Test4 OK, got expected exception %s" % repr(e))
+        print("Test4 OK, got expected exception %s" % repr(e))
+
 
+def test5():
+    ''' NotKillExecTimeoutException
+    '''
     @thread_timeout(1, kill=False)
-    def test5(x):
+    def looong_and_unkillable(x):
         for a in range(0, x):
             time.sleep(2)
     try:
-        test5(2)
+        looong_and_unkillable(2)
+        raise Exception('NotKillExecTimeoutException was expected')
     except NotKillExecTimeoutException as e:
-        print ("Test5 OK, got expected exception %s" % repr(e))
+        print("Test5 OK, got expected exception %s" % repr(e))
+
+
+def test6():
+    ''' decorator is not changing python's into inspection    
+    '''
+    from inspect import getargspec
+
+    def func(x, y=1, *args, **kwargs):
+        return vars()
+
+    func_with_timeout = thread_timeout(1)(func)
+    assert getargspec(func) == getargspec(func_with_timeout)
+
+def test6():
+    ''' Class methods    
+    '''
+    class Class(object):
+
+        @thread_timeout(1) 
+        def short(self, x):
+            return x
+
+        @thread_timeout(1) 
+        def looong(self, x):
+            time.sleep(1000)
```

Comment:
Again, 1000. Think about kittens. You have thread_timeout (1), so 2 is enough to violate timeout restriction.